Disable Quick Edit Mode by default on Windows Consoles, Add runFlag checking to ZMQ and Console threads

claywar opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?
* Disables Quick Edit Mode (Mark) in Windows Consoles to prevent accidental clicks in the window from freezing the server
<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->
* Uses runFlag to signal exit to ZMQ and Console threads to properly exit without SIGABRT
## Steps to test these changes
Windows consoles (cmd, VS debug window) should not register mouse highlight or clicks unless the user right-clicks to "Mark"
<!-- Clear and detailed steps to test your changes here -->

claywar committed:
Disable Quick Edit Mode by default on Windows Consoles


Tabs to spaces


More tabs

zach2good commented:
Could `int runFlag` be changed to `std::atomic<bool> runFlag` or something, not necessarily for safety, but to signal what it is and how it's used

zach2good commented on src/common/console_service.h:
```diff
@@ -19,6 +19,9 @@
 */
 
+#ifndef _CONSOLE_SERVICE_H_
+#define _CONSOLE_SERVICE_H_
```
No need for header guards like this, #pragma once does the same thing, it's just not part of the standard (or something)

claywar replied:
Was failing on GCC10 without these

zach2good commented on src/common/kernel.cpp:
```diff
@@ -46,7 +46,8 @@
 #endif
 
-int    runflag = 1;
+std::atomic<bool> runFlag = true;
```
Maybe rename this gRunFlag? Since it's mega-global

claywar replied:
Will do 👍

claywar committed:
Add runFlag to ZMQ and Console threads, disable SQL Async thread


Missing naming conversion


Test temporary fix to world server


Simplify fix, declare runFlag


Convert runFlag to std::atomic<bool>


Remove unused variable


Fix GCC Build error


Update runFlag variable name to gRunFlag
